ios/browser/api/bookmarks/exporter/bookmarks_encoder.cc

```diff
@@ -0,0 +1,191 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/bookmarks/exporter/bookmarks_encoder.h"
+#include "brave/ios/browser/api/bookmarks/exporter/exported_bookmark_entry.h"
+#include "components/bookmarks/browser/bookmark_codec.h"
+
+#include <stddef.h>
+#include <algorithm>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/guid.h"
+#include "base/json/json_string_value_serializer.h"
+#include "base/memory/ptr_util.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "base/values.h"
+#include "components/bookmarks/browser/bookmark_model.h"
+#include "components/strings/grit/components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "url/gurl.h"
+
+using base::Time;
+
+using bookmarks::BookmarkCodec;
+
+namespace ios {
+// Current version of the file.
+static const int kCurrentVersion = 1;
+
+class BookmarkEncoder {
```

Comment:
why aren't we using BookmarkCodec directly? It doesn't have any chrome dependencies